### ISSUE TITLE
Add gestures for manipulating csproj project files

### ISSUE THREAD

**Issue by DustinCampbell** [OmniSharp/omnisharp-vscode#436]
Currently, manipulating a csproj file can be painful. Adding or removing files and references have to be done manually by editing the project file itself.

**Comment by borgdylan** [comment]
Would this be just for .csproj or will it allow registration of other msbuild based projects?

**Comment by DustinCampbell** [comment]
Currently the plan is just for .csproj. It's related to the work to move over to MSBuild-based .NET Code projects.

**Comment by borgdylan** [comment]
By other, I meant projects that are .NET based but not written in C# i.e. use other compilers that target the CLR/CoreCLR.

**Comment by DustinCampbell** [comment]
No, that's not a goal. This extension only provides C# support for VS Code.

**Comment by borgdylan** [comment]
Would it be possible to have an option to treat other project files as csproj?

**Comment by DustinCampbell** [comment]
Could you be a bit more specific? What sorts of projects are you asking for?

**Comment by borgdylan** [comment]
Csproj like projects that do not have a .csproj extension. A VB project for instance. I am only expecting support for the parts of the file that are like a csproj.

**Comment by borgdylan** [comment]
I.e. in the case of vbproj I do not expect actual VB support, just the help in manipulating the XML for adding dependencies etc.

**Comment by borgdylan** [comment]
Project.json was nice as it had he same file name for all projects regardless of compiler. It's a shame that tools can chose to support just csproj when csproj , vbproj mynetlangproj etc. would all have the same structure and only vary in what language targets file they use.

**Comment by DustinCampbell** [comment]
Unfortunately, this is not in the plans for the C# extension for now. The work to manipulate the csproj file will likely be handled within the OmniSharp process, which specifically only understands C#. That is, the work to manipulate project files will not likely be handled in a language-neutral way by simple manipulation of XML text. For VB support, definitely add your vote to https://github.com/OmniSharp/omnisharp-vscode/issues/25.

**Comment by JoeRobich** [comment]
If these are added, they would be part of the C# DevKit. Also recommend using the dotnet CLI from the terminal.
